Jenkinsfile

```diff
@@ -1,5 +1,92 @@
 #!groovy
 
+{ ->
+    try {
+        node('docker') {
+            stage 'SCM'
+            if (isPullRequest()) {
+                checkout([
+                    $class: 'GitSCM',
+                    branches: [[name: "origin/pull/${GITHUB_PR_NUMBER}/head"]],
+                    doGenerateSubmoduleConfigurations: false,
+                    extensions: [[$class: 'CleanCheckout']],
+                    gitTool: 'native git',
+                    submoduleCfg: [],
+                    userRemoteConfigs: [[
+                    credentialsId: '1642fb1a-1a82-4b10-a25e-f9e95f43c93f',
+                    name: 'origin',
+                    refspec: "+refs/heads/master:refs/remotes/origin/master +refs/pull/${GITHUB_PR_NUMBER}/head:refs/remotes/origin/pull/${GITHUB_PR_NUMBER}/head",
+                    url: 'https://github.com/realm/realm-java.git'
+                    ]]
+                ])
+            } else {
+                checkout([
+                    $class: 'GitSCM',
+                    branches: [[name: '*/master']],
+                    doGenerateSubmoduleConfigurations: false,
+                    extensions: [[$class: 'CleanCheckout']],
+                    gitTool: 'native git',
+                    submoduleCfg: [],
+                    userRemoteConfigs: [[url: 'https://github.com/realm/realm-java.git']]
+                ])
+            }
+
+            stage 'Docker build'
+            def buildEnv = docker.build 'realm-java:snapshot'
+            buildEnv.inside("--privileged -v /dev/bus/usb:/dev/bus/usb -v ${env.HOME}/gradle-cache:/root/.gradle") {
+                stage 'JVM tests'
+                try {
+                    gradle 'assemble check javadoc'
+                } finally {
+                    storeJunitResults 'realm/realm-annotations-processor/build/test-results/TEST-*.xml'
+                }
+                if (env.BRANCH_NAME == 'master') {
```

Comment:
Perhaps move this to the bottom alongside the OJO upload. No need to collect metrics if it fails some of the quality checks?